manager/orchestrator/taskreaper/task_reaper.go

```diff
@@ -282,6 +307,39 @@ func (tr *TaskReaper) tick() {
 			return nil
 		})
 	}
+
+	// Perform service cleanup
+	deleteServices := make(map[string]struct{})
+	// figure out which services need to be cleaned up
+	if len(tr.cleanupServices) > 0 {
+		tr.store.View(func(tx store.ReadTx) {
+			for svc := range tr.cleanupServices {
+				tasks, err := store.FindTasks(tx, store.ByServiceID(svc))
+				// if all tasks of svc have been cleaned up already, then we should get a response
+				// with length zero. If an error occurs though, we log it and don't mark the service
+				// for removal yet.
+				if err != nil {
+					log.G(context.Background()).WithError(err).Errorf("error in task reaper while looking up tasks for service %s", svc)
```

Comment:
lets pass in the context from the task reaper instead of using context.Background()